kernel: mem_manage: un-expose an internal variable and some doxygen fixes

dcpleung opened the pull request:
* Un-expose an internal variable that is only being used in one file.
* Fixes doxygen stuff in `sys/mem_manage.h` so the API doc is more usable.

dcpleung committed:
kernel: mmu: no need to expose z_free_page_count

z_free_page_count is only used in one file, so there is
no need to expose it, even to other part of kernel.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

kartben commented on include/zephyr/sys/mem_manage.h:
```diff
@@ -233,10 +233,10 @@ extern "C" {
- * @param phys Physical address base of the memory region
- * @param size Size of the memory region
- * @param flags Caching mode and access flags, see K_MAP_* macros
+ * @param[out] virt Output virtual address storage location
```
doxygen doesn't "build" the doc for "z_" symbols but you may still want to correct:

```suggestion
 * @param[out] virt_ptr Output virtual address storage location
```

dcpleung replied:
Fixed.

kartben reviewed (commented):
Could you clarify what you mean by "un-expose an internal variable"? All the "z_" symbols are already supposed to be internal, are they not?

dcpleung commented:
> Could you clarify what you mean by "un-expose an internal variable"? All the "z_" symbols are already supposed to be internal, are they not?

It is still declared in the public header, so that is somewhat visible when using IDE. Also, that variable is only used in one file where it can be declared static in that file.

cfriedt reviewed (dismissed):
Looks ok to me

carlescufi commented:
@dcpleung please rebase

dcpleung committed:
kernel: mm: fix some doxygen issues

Fix misplaced in,out in @param. If any parameters are output,
mark the others as input for clarity.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>
